pkg/client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx

```diff
@@ -97,17 +100,57 @@ export const AnalysisWizard: React.FunctionComponent<IAnalysisWizard> = ({
       artifact: "",
       targets: [],
       sources: [],
-      withKnown: "",
+      withKnown: "depsOnly",
       includedPackages: [""],
       excludedPackages: [""],
       customRulesFiles: [],
       excludedRulesTags: [""],
     },
   });
 
-  console.log(methods.watch());
+  const { identities, fetchIdentities } = useFetchIdentities();
+
+  React.useEffect(() => {
+    fetchIdentities();
+  }, [fetchIdentities]);
+
+  const hasIdentity = (application: Application, kind: string): boolean =>
+    !!application.identities?.some((appIdentity) =>
+      identities?.find(
+        (identity) => appIdentity.id === identity.id && identity.kind === kind
+      )
+    );
+
+  const areApplicationsBinaryEnabled = (): boolean =>
+    applications.every(
+      (application) =>
+        application.binary !== "::" &&
+        application.identities &&
+        application.identities.length > 0 &&
+        hasIdentity(application, "maven")
+    );
+
+  const areApplicationsSourceCodeEnabled = (): boolean =>
+    applications.every(
+      (application) =>
+        application.repository &&
+        application.repository.url !== "" &&
+        application.identities &&
+        application.identities.length > 0
+    );
+
+  const areApplicationsSourceCodeDepsEnabled = (): boolean =>
+    applications.every(
+      (application) =>
+        application.repository &&
+        application.repository.url !== "" &&
+        application.identities &&
+        application.identities.length > 0 &&
+        hasIdentity(application, "maven")
+    );
 
   const { handleSubmit, watch, reset } = methods;
+  console.log(watch());
```

Comment:
```suggestion
  React.useEffect(() => {
    const subscription = watch(() => {});
    return () => {
      subscription.unsubscribe();
    };
  }, [watch]);```

Reply:
We should use the recommended usage of watch shown here: https://react-hook-form.com/api/useform/watch/

Reply:
Actually there is no need for watch(), we use it for it's side effects.
If you remember not using it freezed the radio buttons on the `SetScope` page.
Unfortunately setting a subscription doesn't work either as this is creating a circular reference error.
There seems to be limitation of this library with advanced type, which we might be hitting when using string arrays, not sure, but this is really annoying, see  https://github.com/react-hook-form/react-hook-form/issues/4670 and https://github.com/react-hook-form/react-hook-form/issues/4704 discussions about it.
In the meantime we decide what to do I'll revert to console.log(watch).

Reply:
Yeah I am not sure what to do about the circular reference error either. Looks like the creator/maintainer recommends using simple string types for all form value fields until submit time? Not sure if that is even feasible with our application. 

Ok if you do not need it we can remove it entirely. And if we do need it, let's use some better syntax than console.log is all I ask. Like the docs suggest:

const watchAllFields = watch(); // when pass nothing as argument, you are watching everything

Reply:
We need it until we fix the related issue blocking the SetScope component.
I'm fine with a dummy watchAllFields